tools/registry.c

```diff
@@ -1,16 +1,17 @@
 #include <stdio.h>
 #include <stdlib.h>
-#ifdef _WIN32
+#if defined(_WIN32) || defined(__CYGWIN__)
 # include <io.h>
+#endif
+#ifdef _WIN32
 # define rindex(X,Y) strrchr(X,Y)
 # define index(X,Y) strchr(X,Y)
-#else
+#endif
 # include <sys/time.h>
 # include <sys/resource.h>
```

Comment:
I suspect these two headers are still unavailable on Windows; the others might be.